evadb/functions/extract_column.py

```diff
@@ -0,0 +1,101 @@
+# limitations under the License.
+
+import pandas as pd
+
+from evadb.catalog.catalog_type import NdArrayType
+from evadb.functions.chatgpt import ChatGPT
+from evadb.functions.decorators.decorators import forward
+from evadb.functions.decorators.io_descriptors.data_types import PandasDataframe
+
+
+class ExtractColumnFunction(ChatGPT):
+    @property
+    def name(self) -> str:
+        return "EXTRACT_COLUMN"
+
+    def setup(
+        self, model="gpt-3.5-turbo", temperature: float = 0, openai_api_key=""
+    ) -> None:
+        super(ExtractColumnFunction, self).setup(model, temperature, openai_api_key)
+
+    @forward(
+        input_signatures=[
+            PandasDataframe(
+                columns=["field_name" "description", "data_type", "input_rows"],
+                column_types=[
+                    NdArrayType.STR,
+                    NdArrayType.STR,
+                    NdArrayType.STR,
+                    NdArrayType.STR,
+                ],
+                column_shapes=[
+                    (1,),
+                    (1,),
+                    (1,),
+                    (1,),
+                ],
+            )
+        ],
+        output_signatures=[
+            PandasDataframe(
+                columns=["response"],
+                column_types=[
+                    NdArrayType.STR,
+                ],
+                column_shapes=[(1,)],
+            )
+        ],
+    )
+    def forward(self, unstructured_df):
+        """
+        NOTE (QUESTION) : Can we structure the inputs and outputs better
+        The circumvent issues surrounding the input being only one pandas dataframe and output columns being predefined
+        Will add all column types as a JSON and parse in the forward function
+        Provide only the file name from which the input will be read
+        Output in JSON which can be serialized and stored in the results column of the DF
+        """
+        field_name = unstructured_df.iloc[0, 0]
+        description = unstructured_df.iloc[0, 1]
+        data_type = unstructured_df.iloc[0, 2]
+        input_rows = unstructured_df[unstructured_df.columns[3]]
+        prompt = """
+            You are given a user query. Your task is to extract the following fields from the query and return the result in string format.
+            IMPORTANT: RETURN ONLY THE EXTRACTED VALUE (one word or phrase). DO NOT RETURN THE FIELD NAME OR ANY OTHER INFORMATION.
+        """
+        content = """
+            Extract the following fields from the unstructured text below:
+            Format of the field is given in the format
+            Field Name: Field Description: Field Type
+            {}: {}: {}
+            The unstructured text is as follows:
+        """.format(
+            field_name, description, data_type
+        )
+
+        print(prompt)
+        print(content)
+
+        output_df = pd.DataFrame({"response": []})
+
+        for row in input_rows:
```

Comment:
We shall not use the first field_name, description, data_type  for all the rest rows. 
```python
for row in unstructured_df.itertuples():
    field_name = row[0]
    description = row[1]
    data_type = row[2]
    input_rows = row[3]

    ...
```